### ISSUE TITLE
[Feature] Option to see(/change/clear) if PullSecret is store in credential manager (Windows)

### ISSUE THREAD

**Issue by gbraad** [crc-org/crc#3832]
Currently a user is unable to verify and/or change the pull-secret that is stored in the Credential Manager. This can lead to confusion why the pullsecret has not be requested, or difficult to modify if a changed pullsecret is needed.

Ref: https://github.com/crc-org/crc-extension/issues/171

**Comment by anjannath** [comment]
I think for the extension's use case, we are good with only adding this capability to the API, do we want a CLI interface also to see the stored pull-secret and remove it?

currently `crc cleanup` removes the stored pull-secret, but do we want a specific command to see/remove the pull-secret?

**Comment by gbraad** [comment]
'see' might be more like 'is set', like we have already by API but visual from the config maybe?

**Comment by rohanKanojia** [comment]
I’ve made some changes to add a new option in CRC config named `pull-secret-stored-keyring` (defaults to `false`)

If user starts the CRC cluster with `crc start` ,this config option would be set to true if CRC is able to store the fetched pull-secret into Operating System keyring.

When we would run `crc config view`, we should be able to see this config flag option in the output like this:
```shell
crc : $ crc config view
- consent-telemetry                     : yes
- pull-secret-stored-keyring            : true
```

I am not 100% sure whether this is what was expected or I misunderstood the issue. Now that I'm testing it I'm confused whether there might be a better way of showing this to user rather than a config option. CRC config stores user preferences with respect to the cluster he would start. I'm not sure whether adding a config option just to show whether pull-secret is cached or not is the right way to inform user about pull-secret.

**Comment by cfergeau** [comment]
When the pull secret is passed on the command line, or is set in the config, does it take precedence over the one set in the credential manager? When the user specifies the pull secret, I would use this and replace the one which is in the credential manager.

**Comment by rohanKanojia** [comment]
> When the pull secret is passed on the command line, or is set in the config, does it take precedence over the one set in the credential manager? 

Yes, it seems to be the case. On looking at [pullsecret.go](https://github.com/crc-org/crc/blob/main/pkg/crc/cluster/pullsecret.go#L78)'s nonInteractivePullSecretLoader:
https://github.com/crc-org/crc/blob/faefe124ef71902172238ee8fe62b12e518038e0/pkg/crc/cluster/pullsecret.go#L84-L90

I see the following precedence order for loading the pull secret value:
1. Pull Secret from Path `--pull-secret-file`
2. Pull Secret from config `pull-secret-file`
3. Pull Secret from OS Credential Manager

We can see which source is getting used for choosing pull secret on enabling the debug logs.

> When the user specifies the pull secret, I would use this and replace the one which is in the credential manager.

Do you mean when we detect pull secret from other source we should modify credential manager's pullsecret value. At the moment we only update the value when we issue prompt for pull secret:

https://github.com/crc-org/crc/blob/faefe124ef71902172238ee8fe62b12e518038e0/pkg/crc/cluster/pullsecret.go#L48-L58

**Comment by cfergeau** [comment]
> Do you mean when we detect pull secret from other source we should modify credential manager's pullsecret value. 

When the user stores it in the config/removes it from the config, imo it makes sense to sync it to the keyring/remove it (though we only store the path to a file I believe?).
When it's specified on the command line, not so sure we should do it, it's more temporary/transient, so maybe better not to store anything

**Comment by rohanKanojia** [comment]
> When the user stores it in the config/removes it from the config, imo it makes sense to sync it to the keyring/remove it (though we only store the path to a file I believe?).

Yes, you're right. We only store the path in config option and complete pull secret content in keyring. I can update code to also store pull secret in keyring by reading contents of file specified in `pull-secret-file` config option.  However I'm not sure how to deal with the removing pullsecret from keyring when `pull-secret-file` is unset (need to check it).

> When it's specified on the command line, not so sure we should do it, it's more temporary/transient, so maybe better not to store anything

👍  I think these options are for one-off overrides. In my opinion, we shouldn't store anything in this case as you suggested.

**Comment by praveenkumar** [comment]
> When the pull secret is passed on the command line, or is set in the config, does it take precedence over the one set in the credential manager?

Yes, it take the precedence over keyring.

> When the user specifies the pull secret, I would use this and replace the one which is in the credential manager.

As per code we only store the pull secret when it is prompt to the user. 
- If pull-secret-file config option or start flag in used it is not stored in keyring

This means we are not going to prompt user until they remove the pull-secret-file from config and clean the keyring. The whole point was to store in keyring so that user don't get prompted every single time and since we now remove that as part of `cleanup`, user can update the pull-secret.

**Comment by rohanKanojia** [comment]
I had an internal discussion about this issue with Praveen.

_originally posted by @praveenkumar in internal chat:_

Regarding whether we should store pullsecret in keyring whenever it's provided via config option:
> - But when user is specify the pull-secret-file as part of config then why we want to save it on keyring because that file is already there and that is precedence over keyring?
> - I don't think we need anything now because this was created when we were not cleaning up pull-secret from keyring as part of cleanup command but since now we have that, there is already way to update it as long as user prompt for it.


We log debug messages to show from which source pull secret is getting loaded. We have decided to keep it like this. 

I don't think we need to do anything extra in scope of this issue.
